### PR TITLE
Expose Byline to DCR

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Export `byline` to the `DotcomRenderingDataModel`.

Part of #25220 

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [X] Yes TBC

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala [hunk 1]
```diff
@@ -484,7 +488,7 @@ object DotcomRenderingDataModel {
       tags = dcrTags,
       trailText = TextCleaner.sanitiseLinks(edition)(content.trail.fields.trailText.getOrElse("")),
       twitterData = page.getTwitterProperties,
-      version = 3,
+      version = 4,
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:491]: Maybe worth bumping once we remove `author`

**Review by jamesgorrie — APPROVED** [review]
Are we going to want `bylineHtml`, or will we still construct those in DCR?

**Comment by mxdvl** [comment]
> Are we going to want `bylineHtml`, or will we still construct those in DCR?

For the time being, I think it’s best to handle that in DCR. If we could get a contract of how structured the data is, it might be useful for us to parse it, but I think that’s a subsequent step.

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (mxdvl) [reply common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:491]:
```suggestion
      version = 3,
```

**Review by jlieb10 — APPROVED** [review]

**Pull request merged**